/***
 *  @file : Exception.hpp
 * 	@date 2019/08/18
 *  @author 하정현(seokbong60@gmail.com) Team:SweetCase Project (1-person Team)
 *  @brief : 서버와 클라이언트, 스레드 간의 겍체 통신을 할 때 사용합니다.
 * 
 ***/

syntax = "proto3";

package PacketSerialData;

//DataElement
message DataElementSerial {
    string data = 1;
    int32 dataType = 2;
    int32 structType = 3;
}

//Packet
message PacketSerial {
    int32 packetType = 1;
    string username = 2;
    string IP = 3;
    int32 cmdNum = 4;
    int32 sock = 5;
}

//SendCmdPacket
message SendCmdPacketSerial {
    PacketSerial packet = 1;
    repeated string cmdVector = 2;
}

//RecvPacket
message RecvPacketSerial {
    PacketSerial packet = 1;
    int32 recvPacketType = 2;
}

//recvDataPacket
message RecvDataPacketSerial {
    RecvPacketSerial recvPacket = 1;
    DataElementSerial data = 2;
}

//RecvMsgPacket
message RecvMsgPacketSerial {
    RecvPacketSerial recvPacket = 1;
    string msg = 2;
}

//SignalPacket
message SignalPacketSerial {
    PacketSerial packet = 1;
    int32 signal = 2;
}

//LogPacket
message LogPacketSerial {
    PacketSerial packet = 1;
    string dateformat = 2;
    string str = 3;
}
